protostar/commands/test/environments/factory.py

```diff
@@ -0,0 +1,22 @@
+from typing import Optional
+
+from protostar.commands.test.environments.setup_execution_environment import (
+    SetupExecutionEnvironment,
+)
+from protostar.commands.test.environments.test_execution_environment import (
+    TestExecutionEnvironment,
+)
+from protostar.commands.test.execution_state import ExecutionState
+from protostar.commands.test.starkware import ExecutionResourcesSummary
+
+
+async def invoke_setup(function_name: str, state: ExecutionState):
+    env = SetupExecutionEnvironment(state)
+    await env.invoke(function_name)
+
+
+async def invoke_test_case(
+    function_name: str, state: ExecutionState
+) -> Optional[ExecutionResourcesSummary]:
+    env = TestExecutionEnvironment(state)
```

Comment:
FYK: I try to move class instantiating as high as possible in the project hierarchy (except dataclasses, POJOs) in order to create `composition_root.py` at some point and maybe integrating DI library.

Reply:
DI is useful to make the classes easy to unit test and easier to understand what is dependent on what.

Reply:
In case of test runner integration tests make more sense hence DI is not that important. Resolving conversation.